sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/command/AlterTableDropPartitionSuite.scala

```diff
@@ -39,11 +39,11 @@ class AlterTableDropPartitionSuite
           checkHiveClientCalls(expected = if (statsOn) 26 else 18) {
             sql(s"ALTER TABLE $t DROP PARTITION (part=2)")
           }
-          checkHiveClientCalls(expected = if (statsOn) 31 else 18) {
+          checkHiveClientCalls(expected = if (statsOn) 32 else 18) {
```

Comment:
Why stats changed after this pr?

Reply:
Before this PR，when updating table stats, spark will convert CatalogTable to hive table;
After this PR, when updating table stats, spark will get the hive table from hive metastore and then update table stats. Refer to https://github.com/apache/spark/pull/38495/files#diff-45c9b065d76b237bcfecda83b8ee08c1ff6592d6f85acca09c0fa01472e056afR616-R618